src/qt/bitcoinamountfield.cpp

```diff
@@ -326,14 +326,14 @@ void BitcoinAmountField::unitChanged(int idx)
     unit->setToolTip(unit->itemData(idx, Qt::ToolTipRole).toString());
 
     // Determine new unit ID
-    int newUnit = unit->itemData(idx, BitcoinUnits::UnitRole).toInt();
+    BitcoinUnit newUnit = unit->itemData(idx, BitcoinUnits::UnitRole).value<BitcoinUnit>();
```

Comment:
Why is this safe to do in qt? Any integer can be converted to `BitcoinUnit`, at least in C++.

Reply:
I believe, no `int` to `BitcoinUnit` conversion here. [`QComboBox::itemData()`](https://doc.qt.io/qt-5/qcombobox.html#itemData) does a kind of mapping from `int` index to `QVariant`.

Anyway, going to make this code more safer.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/17877#issuecomment-636514466).